test_conformance/select/util_select.cpp

```diff
@@ -242,6 +240,28 @@ void refselect_1u64u(void *dest, void *src1, void *src2, void *cmp, size_t count
         d[i] = m[i] ? y[i] : x[i];
 }
 
+void refselect_hhi(void *dest, void *src1, void *src2, void *cmp, size_t count)
+{
+    size_t i;
+    cl_short *d, *x, *y, *m;
+    d = (cl_short *)dest;
```

Comment:
This would be better written as 

`const cl_short *const d = (cl_short *)dest;`

1 less line and less opportunity for error, in the distant future.  Same for all, maybe to be like existing code I guess? Which I could live with.

Reply:
My view others may differ is that if we make a change we make it as good (within reason) as possible. Sometimes the event in the future happens, and sometimes it doesn't.  So presume it doesn't and we always get code at its best.

Reply:
Thanks Chris, let's clarify first one concern. Didn't you mean:

`cl_short *const d = (cl_short *)dest;`

with your additional qualifier `const` build of my project (under Ubuntu) fails.

Reply:
On the refactoring point, this is the kind of "modernization" I was thinking of from the OpenCL call discussion. Getting rid of old C89 conventions in the codebase that split variable definition & declaration, rather than using modern C++ features.

We suggested pushing separate PRs for these fp16 changes - A first one with the functional additions for fp16, and a second with the clean-up changes. However, what we could do is make them a single PR, but logically split the commits. So that the first commit is the functional addition, and a second commit is refactoring clean-ups, and later commits to address PR feedback. Then reviewers can look at the commits individually to understand the changes better, but the whole PR would be the best possible code.

However, I don't think we should start pushing clean-up changes to this PR in particular right now, until the in progress work by Holochip on the same test is merged (or even pushed), since we don't want conflicts and the work to be wasted.

Reply:
@MeyeChris I've went through all similar cases and added requested consts